Ajax Skeleton

BenGitsCode opened the pull request:
Built basic skeleton for Ajax requests and associated files.
- Auth Folder containing:
- ui.js  Success and Failure handlers
- events-user.js Click events for user
- api-user.js Ajax calls for user

In Scripts:
- app-data.js for use in above scripts

BenGitsCode committed:
Created Auth folder
- ajax-user.js
- api.js
- app-data.js

BenGitsCode committed:
Created Events-user.js and ui.js
- user click events
  - sign up
  - sign in
  - sign out
  - change password

BenGitsCode committed:
UI.js skeleton
- also moved app-data into scripts

BenGitsCode committed:
Deleted irrelevant comment

BenGitsCode committed:
Deleted ajax-user.js
- Will put user ajax calls in api-user.js
- More clarity, Team approved

BenGitsCode committed:
Renamed api to api-user
- Better modularity

BenGitsCode committed:
Api-user.js Ajax calls created
- signUp
- signIn
- signOut
- changePass

BenGitsCode commented:
I realize https://github.com/PizzaBeer/filebucket-front-end/pull/1/commits/34fc36d49d3a551fee3dc1d0366c75e5e8607d20 did not have a comment. I should have ammended a correction.

cuprous commented on assets/scripts/app-data.js:
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+const app = {
+  api: 'http://localhost:8080/',
```
API isn't on localhost:8080; it's going to be localhost:3000.

BenGitsCode replied:
Corrected.

cuprous commented on assets/scripts/auth/events-user.js:
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const getFormFields = require('../../../lib/get-form-fields.js');
```
I think getFormFields is globally available in this template.

BenGitsCode committed:
Changed api to http://localhost:3000/
-in app-data.js

cuprous commented on assets/scripts/auth/events-user.js:
```diff
@@ -0,0 +1,42 @@
+
+const getFormFields = require('../../../lib/get-form-fields.js');
+
+const authApi = require('./api');
```
This should be ./api-user, no?

BenGitsCode replied:
Yep! Missed that after I renamed api.js Corrected and pushed.

cuprous commented on assets/scripts/auth/ui.js:
```diff
@@ -0,0 +1,31 @@
+  success,
+  signOutSuccess,
+  signInSuccess,
+  app,
```
Don't think you need to export app here.

natdjerf replied:
Instead of 

``` js
app.user = data.user;
```

I think it would be helpful to store user information in an object on the app file. On app, we would have to update the export to include app & currentUser, and on the ajax calls, app.api & app.currentUser

``` js
let currentUser = {
  id: undefined,
  token:''
};
```

I think it will be helpful for some of the handlebars rendering and to avoid circular references on the JS files.

BenGitsCode replied:
Changed to:

``` js
let currentUser = {
  id: undefined,
  token:''
};

const signInSuccess = (data) => {
  app.user = currentUser;
  console.log(app);
  console.log(data);
  console.log("Sign in successful");
};
```

Also added a console.log of data

cuprous commented on assets/scripts/auth/api-user.js:
```diff
@@ -0,0 +1,60 @@
+  signIn,
+  signOut,
+  changePass,
+  app,
```
Don't think you need to export app.

rdelvallej32 commented on assets/scripts/auth/events-user.js:
```diff
@@ -0,0 +1,42 @@
+const authUi = require('./ui');
+
+
+const signCheck = () => {
```
rename to maybe eventHandlers?

BenGitsCode committed:
changed signCheck to userEvents
- more semantic in context
- need to double check it getFormFields is globally available

cuprous commented on assets/scripts/auth/api-user.js:
```diff
@@ -0,0 +1,60 @@
+    data,
+
+  })
+  .done(success)
```
Console log these via fat arrow functions for now.

BenGitsCode replied:
Could  you clarify @cuprous

cuprous replied:
A few things:
- success and failure shouldn't be something that is passed to signIn() in the first place.
- line 16: change to const signIn = (data) => {
- line 23: change to }).done((success) => console.log(success))
- line 24: change to .fail((fail) => console.error(fail));

Basically, change the things in .done and .fail to callbacks, kind of like with promises.

BenGitsCode replied:
Made the changes as shown in 
https://github.com/PizzaBeer/filebucket-front-end/commit/3cf0c94b7441802b6242efe5aacc81e675f2cfc0

How does that look?

BenGitsCode committed:
Removed app from exports in api-user.js
-Not Necessary

BenGitsCode committed:
Modified instances of app.user into currentUser
   const signInSuccess = (data) => {
-  app.user = data.user;
+  app.user = currentUser;

BenGitsCode committed:
Corrected authApi require
-Changed from ./api to ./api-user
-did not update in accordance with file name change
from api to api-user

BenGitsCode committed:
Modified success and fails on ajax
- changed:
  .done(success)
  .fail(failure);
};

to:   .done((success) => console.log(success))
   .fail((fail) => console.error(fail));
  };

-changed:
-const signIn = (success, failure, data) => {

to: const signIn = (data) => {

cuprous commented on assets/scripts/auth/api-user.js:
```diff
@@ -0,0 +1,58 @@
+
+const app = require('../app-data');
+
+const signUp = (success, failure, data) => {
```
Same comments as signIn. Fix signOut, too.

BenGitsCode replied:
Modified accordingly. signOut did not have data in the ajax so I added that as well.

BenGitsCode committed:
changed signout and signup parameters to data
-data added into signout ajax

BenGitsCode committed:
Added data into Signout ajax
